### DISCUSSION TITLE
Immersive voice chat tweak

### DISCUSSION THREAD

**Discussion by cherrydrum** [FakeFishGames/Barotrauma#12718] [Ideas and suggestions]
The in-game voice chat is a godsend. The applied audio effects, the effective range of radio communication is contributing to the claustrophobic atmosphere immensely. There is one thing that would make it even better though.

The VC controls is still a bit clunky. It would be wonderful to have an option to make voice radio messages push-to-talk, _while keeping the local chat being set to voice activity_ This way the players will be able to talk to each other casually when they are close to each other, but at the same time will have to push a separate button to use radio comms.

In my personal experience, when the crew size becomes larger than five people, radio comms become a complete mess, and most of the clutter is generated unintentionally, simply because the players want to discuss something, but forget to press a button (why?) to switch to local chat.

With this tweak, the communications will be more effective, and, which is more cool, immersive.

I've already looked up the lua modding documentation and failed to implement it by myself, but hope never dies.

P.S. Found a similar idea (#9039), updating because trying to prove why it deserves attention. The solution from this discussion does not work anymore.  ```uselocalvoicebydefault```  is nowhere to be found.

**Comment by cherrydrum** [comment]
Found a way to implement this using legacy keybinds. But why is this option legacy?

**Comment by Hazmat616** [comment]
This makes sense because in real life you just talk up close and must push a button to use a radio.

**Comment by bookslayer10** [comment]
It would be nice if there was one button to push to talk on local VC, and a different button to push to talk on radio VC. Then, options to make those toggles so you don't need to hold them down.

**Comment by cherrydrum** [comment]
> It would be nice if there was one button to push to talk on local VC, and a different button to push to talk on radio VC. Then, options to make those toggles so you don't need to hold them down.

This feature is already implemented. You can find it in your control settings. It's marked as legacy, though.
